spark-cassandra-connector/src/main/scala/com/datastax/spark/connector/util/ReflectionUtil.scala

```diff
@@ -85,13 +85,31 @@ object ReflectionUtil {
     getters(implicitly[TypeTag[T]].tpe)
   }
 
-  /** Returns the type of the parameters of the given method.
+  /** Looks up a method by name in a type.
     * The method must exist, otherwise `IllegalArgumentException` is thrown. */
-  def methodParamTypes(tpe: Type, methodName: String): Seq[Type] = TypeTag.synchronized {
+  def method(tpe: Type, methodName: String): Type = TypeTag.synchronized {
     val member = tpe.member(newTermName(methodName))
```

Comment:
I don't think ReflectionUtil.scala is private[connector] so consider things like this, where the function is public to the user API but there are no checks on the args passed in, should have a require, for example require(methodName != null || methodName.nonEmpty, "'methodName' must have a valid value.")